### PR TITLE
Release 3.5.0

### PR DESCRIPTION
**Pull request by konraddysput**
# Backtrace-Unity@3.5.0 release

Changes:
- Metrics support,
- Breadcrumbs support,
- Fixed Unity 2020, Unity 2021 warnings,
- Adjusted structure size on the iOS build
- Guess backtrace-android native crash reporter path if appcontext is not available

### CODE REVIEW

File: Runtime/Model/Breadcrumbs/Storage/BacktraceStorageLogManager.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public long BreadcrumbsSize
         /// <summary>
         /// Default breacrumbs size. By default breadcrumbs file size is limitted to 64kB.
```
Comment (vlussenburg) [inline Runtime/Model/Breadcrumbs/Storage/BacktraceStorageLogManager.cs:41]: Change to `Default bredacrumbs size, by default limited to 32kB.`

**Review by rqbacktrace — APPROVED** [review]
Already reviewed and approved these new features separately, looking forward to the new release!

**Review by vlussenburg — APPROVED** [review]
See test report in Hack.

File: Tests/Runtime/Metrics/BacktraceMetricsTests.cs [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+            };
+            var backtraceMetrics = new BacktraceMetrics(_attributeProvider, 0, _defaultUniqueEventsSubmissionUrl, _defaultSummedEventsSubmissionUrl);
+            backtraceMetrics.OverrideHttpClient(requestHandler);
+
+            backtraceMetrics.AddUniqueEvent(UniqueAttributeName);
+            backtraceMetrics.Send();
+
+            Assert.IsNotEmpty(jsonString);
+            Assert.AreEqual(submissionUrl, _expectedUniqueEventsSubmissionUrl);
+            Assert.IsEmpty(backtraceMetrics.SummedEvents);
```
Comment (rqbacktrace) [inline Tests/Runtime/Metrics/BacktraceMetricsTests.cs:96]: Should this be checking that `backtraceMetrics.UniqueEvents` is empty too?

File: Runtime/Model/Metrics/UniqueEvent.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using Backtrace.Unity.Json;
+using System.Collections.Generic;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Model.Metrics
+{
+    public sealed class UniqueEvent : EventAggregationBase
```
Comment (rqbacktrace) [inline Runtime/Model/Metrics/UniqueEvent.cs:7]: When do we clear UniqueEvents?

**Review by rqbacktrace — COMMENTED** [review]
Does UniqueEvents clear properly?

File: Runtime/Model/Metrics/UniqueEvent.cs (hunk 3 above)
Comment (rqbacktrace) [reply Runtime/Model/Metrics/UniqueEvent.cs:7]: I see...we just update the timestamp of all unique events we are tracking, thank you for explanation @konraddysput

**Comment by konraddysput** [comment]
> Does UniqueEvents clear properly?

Unique events stay in the application memory - after successful send, backtrace-unity refresh timestamp

**Pull request merged**